tests/containers/validate_btrfs.pm

```diff
@@ -65,15 +66,14 @@ sub _test_btrfs_device_mgmt {
     script_run("df -h");
     # Determine the remaining size of /var
     my $var_free = script_output('df 2>/dev/null | grep /var | awk \'{print $4;}\'');
-    # /var is using its own partition. Create file in the container enough to fill up the partition up to 99%
-    my $fill = int($var_free * 0.99 * 1024);    # df returns the size in KiB
+    # Create file in the container enough to fill the "/var" partition (where the container is located) up to 90%
+    # Note: Don't fill it up more than 90% otherwise `btrfs balance` might stumble.
+    my $fill = int($var_free * 1024 * 0.90);    # df returns the size in KiB
     $rt->up('huge_image', keep_container => 1, cmd => "fallocate -l $fill bigfile.txt");
-    validate_script_output "df -h --sync|grep var", sub { m/\/dev\/vda.+\s+(9[7-9]|100)%/ };
+    validate_script_output "df -h --sync|grep var", sub { m/\/dev\/vda.+\s+(9[0-9]|100)%/ };
     # partition should be full
     my ($total, $used) = _btrfs_fi("/var");
-    die "partition should be full" unless (int($total) == int($used));
-    # Due to disk space this should fail
-    die("pull still works") if ($rt->pull("$container") == 0);
+    die "partition utilisation should be > 90%" unless (int($used) >= int($total * 0.90));
     assert_script_run "btrfs device add /dev/vdb $dev_path";
```

Comment:
This is a bit unclear. So the `/dev/sdb` is the additional device we are adding to the btrfs container which contains the docker image and which is mared as `$dev_path`? Are we sure that this device will be always `/dev/sdb`?

Reply:
I don't know about the `/dev/vdb` part of the test run. As far as I can see, this disk has been created elsewhere and not within this test. I've only been fixing the failing `btrfs balance` part without touching this part.
Perhaps @b10n1k knows more about the design of this test case.

Reply:
> This is a bit unclear. So the `/dev/sdb` is the additional device we are adding to the btrfs subvolume which contains the docker images and which is mared as `$dev_path` so usually `/var/lib/docker`? Are we sure that this device will be always `/dev/sdb`?

AFAIK because it is a VM and as far the setting have "HDDMODEL" : "virtio-blk" the device will be presented as `/dev/vdb`. i am not sure if i answered your question tho.

Reply:
As far as the VM is QEMU I understand :)